src/go-socks5/socks5.go

```diff
@@ -161,11 +198,22 @@ func (s *Server) ServeConn(conn net.Conn) error {
 	request.RemoteAddr = &AddrSpec{IP: remoteAddr.IP, Port: remoteAddr.Port}
 
 	// Process the client request
-	if err := s.handleRequest(request, conn); err != nil {
+	if err := s.handleRequest(request, wrappedConn); err != nil {
 		err = fmt.Errorf("Failed to handle request: %v", err)
 		s.config.Logger.Printf("[ERR] socks %s: %v", remoteAddr, err)
 		return err
```

Comment:
We should continue to log the connection even if an error happens mid-way.

Reply:
defer the log writing at new commit.